Podcast/ChangeUsernameView.swift

```diff
@@ -105,13 +110,19 @@ class ChangeUsernameView: UIView, UITextFieldDelegate {
         successView.image = #imageLiteral(resourceName: "success_icon")
         successView.isHidden = false
         continueButton.isEnabled = true
-        continueButton.backgroundColor = .rosyPink
+        continueButton.backgroundColor = .mutedBlue
     }
     
     func changeUsernameFailure() {
+        usernameTextField.isEnabled = true
         successView.image = #imageLiteral(resourceName: "failure_icon")
         successView.isHidden = false
-        usernameTextField.isEnabled = true
+        continueButton.isEnabled = false
+        continueButton.backgroundColor = .silver
+    }
+    
+    @objc func textFieldDidChange(_ textField: UITextField) {
+        delegate?.changeUsernameViewTextFieldDidChange(changeUsernameView: self, username: textField.text!)
```

Comment:
does this throw errors if there's no text/someone deletes everything? or is it just the empty string

Reply:
Hmm I'm not sure, I can try to change `username: textField.text!` to `username: textField.text ?? ""` to safely unwrap it?

Reply:
We did force unwrap it in `textFieldDidEndEditing` so I think it might be okay.
@natashaarmbrust ??

Reply:
i think its fine then if it was used before!

Reply:
safely unwrap it and only call the delegate if it isn't nil and not the empty string